corehq/apps/api/domain_metadata.py

```diff
@@ -30,6 +32,21 @@ def get_truth(inp, relate, cut):
         return ops[relate](inp, cut_datetime)
 
 
+class DomainQuerySetAdapter(object):
+
+    def __init__(self, es_query):
+        self.es_query = es_query
+
+    def count(self):
+        return self.es_query.size(0).run().total
+
+    def __getitem__(self, item):
+        if isinstance(item, slice):
+            doc_ids = self.es_query.start(item.start).size(item.stop - item.start).run().doc_ids
+            return map(Domain.wrap, iter_docs(Domain.get_db(), doc_ids))
```

Comment:
@kkrampa I believe that the documents you get from elasticsearch will be exactly the same as the ones you get from couchdb. (The whole doc is copied into ES.) So if you just call `Domain.wrap` on the docs from ES, you'll end up with the same result.